execute/report/report.go

```diff
@@ -41,6 +38,12 @@ func buildSingleChainReportHelper(
 		}
 	}
 
+	numMsg := len(report.Messages)
+	if len(report.TokenData) != numMsg {
```

Comment:
Just double checking - for tokens that have no offchain token data, it'll just be an empty byte slice?

Reply:
Yes, it also pads the slice in case messages are skipped and/or checked out of order for some reason. Added a couple more test cases for this.